scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -276,16 +283,34 @@ static void dispatch_local(struct task_struct *p, u64 enq_flags)
 }
 
 /*
- * Dispatch a task on a target per-CPU FIFO.
+ * Dispatch a task on a target CPU.
+ *
+ * If it's not possible to dispatch on the selected CPU, try to re-use the
+ * previously used one, if that one is also not usable dispatch to the global
+ * DSQ.
  */
-static void dispatch_on_cpu(struct task_struct *p, s32 cpu, u64 enq_flags)
+static void dispatch_task(struct task_struct *p, s32 cpu, u64 enq_flags)
 {
-	/*
-	 * If it's not possible to dispatch on the selected CPU, re-use the
-	 * previously used one.
-	 */
-	if (!bpf_cpumask_test_cpu(cpu, p->cpus_ptr))
+	if (!bpf_cpumask_test_cpu(cpu, p->cpus_ptr)) {
 		cpu = scx_bpf_task_cpu(p);
+		if (!bpf_cpumask_test_cpu(cpu, p->cpus_ptr)) {
+			/*
+			 * Both the designaged CPU and the previously used CPU
```

Comment:
designated

Reply:
...and fixed! I should really run a syntax checker. And something that checks when I confuse scx_rustland with scx_userland as well. :rofl: 

Thanks!